### PR TITLE
POSA light client 

### PR DESCRIPTION
**Pull request by hujw77**
Refer: https://www.notion.so/itering/POSA-Light-Client-429f2fe8f76c4d4e819b153a247e2c8e

### CODE REVIEW

File: contracts/bridge/src/spec/POSACommitmentScheme.sol [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "../utils/ScaleCodec.sol";
+
+contract POSACommitmentScheme {
+    using ScaleCodec for uint32;
```
Comment (AurevoirXavier) [inline contracts/bridge/src/spec/POSACommitmentScheme.sol:9]: Why we need scale here?

Comment (hujw77) [reply contracts/bridge/src/spec/POSACommitmentScheme.sol:9]: It's better to use abi.encode here.

File: contracts/bridge/src/utils/ECDSA.sol [hunk 2]
```diff
@@ -7,24 +7,53 @@ pragma solidity 0.7.6;
 /// These functions can be used to verify that a message was signed by the holder
 /// of the private keys of a given address.
 library ECDSA {
+
     /// @dev Returns the address that signed a hashed message (`hash`) with
     /// `signature`. This address can then be used for verification purposes.
-    ///
+    //
```
Comment (AurevoirXavier) [inline contracts/bridge/src/utils/ECDSA.sol:13]: Interesting change.

Comment (hujw77) [reply contracts/bridge/src/utils/ECDSA.sol]: It is a mistake. 😄

**Review by HackFisher — APPROVED** [review]

**Pull request merged**